docs: fix usage.yaml in ingress user guide

carslen opened the pull request:
Changed container image in `usage.yaml` to get [ingress user guide](https://kind.sigs.k8s.io/docs/user/ingress/) working again. Old image `hashicorp/http-echo` seems to have problems, new image `azuredocs/aks-helloworld` works fine.

![image](https://user-images.githubusercontent.com/28647218/207697580-633ddd15-af3a-49c9-894b-3d4b04c8f19e.png)

![image](https://user-images.githubusercontent.com/28647218/207703339-cf3afa17-fe0a-4fbe-8b32-65b76bcdd226.png)

resolves #3024

BenTheElder commented:
Thanks for the PR! would prefer https://github.com/kubernetes/website/pull/37383

BenTheElder commented:
I'd also explicitly prefer not to use a branded hello-world, @aojea and I work for GCP but we wouldn't put a GCP specific hello-world here either :+)

We have an image maintained by upstream Kubernetes SIG Testing that can do http echo, which we fixed in the minikube kubernetes.io docs, I forgot to check back on the KIND hello-world for the same issue. We can use the same fix, linked above and in #3024.

Thanks again for looking into this and filing a bug and fix.

carslen commented:
I will look into it :)

carslen commented:
Implemented image `agnhost` instead of `aks-helloworld`, adjusted documentation to changed output. Indentation adjusted in `usage.yaml`.

<img width="668" alt="image" src="https://user-images.githubusercontent.com/28647218/207828360-761cf071-cd3f-4f53-ae16-d74b685f2355.png">

<img width="668" alt="image" src="https://user-images.githubusercontent.com/28647218/207830169-23ace764-c672-467a-8e65-cb426697537e.png">

stmcginnis commented:
/retest

Failures do not look related.

Thanks for the update. Not as pretty, but the right call.

BenTheElder commented on site/content/docs/user/ingress.md:
```diff
@@ -165,8 +165,8 @@ kubectl apply -f {{< absURL "examples/ingress/usage.yaml" >}}
-curl localhost/bar
+# should output "foo-app"
+curl localhost/foo/hostname
+# should output "bar-app"
```
we can iterate on this, but echo?msg=foo and echo?msg=bar seem more appropriate.

carslen replied:
Hm, what's the aim of the curl command? To receiving the identity of these two pods, don't we? Instead, _echo_ will respond with everything you provide to _echo_.

```shell
(kind-kind:default)
[~]$ curl http://localhost/foo/echo\?msg\=bar
bar%
(kind-kind:default)
[~]$ curl http://localhost/foo/hostname
foo-app
```

But the bottom line is that I don't care (it's not meant to be rude). If you want me to add the echo instead of the hostname, that's fine with me :)

BenTheElder replied:
good point
